### PR TITLE
Remove py2 code

### PR DESCRIPTION
**Pull request by mxr**
Related to #225

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -37,6 +36,11 @@ repos:
     - from __future__ import print_function
     - --remove-import
     - from __future__ import unicode_literals
+- repo: https://github.com/asottile/pyupgrade
+  rev: v3.12.0
+  hooks:
+  - id: pyupgrade
+    args: [--py37-plus]  # based on setup.cfg classifiers
```
Comment (macisamuele) [inline .pre-commit-config.yaml:43]: I'm not comfortable with this change as syntax between Python 3.5/3.6/3.7 is slightly different.
Given that this PR aims to drop Python 2 support (as specified in the package setup) I would not impose the fact that at least python 3.7 needs to be used.

Is the proposal of this PR to drop all the Python version that are in EOL (https://devguide.python.org/versions/)?
If so what is the rationale?
If supporting a "deprecated" version has no cost, what is the advantage of removing the support by introducing a different syntax?

File: language_formatters_pre_commit_hooks/__init__.py [hunk 2]
```diff
@@ -11,12 +10,8 @@ def _get_default_version(tool_name: str) -> str:  # pragma: no cover
     The method is intended to be used only from language_formatters_pre_commit_hooks modules
     """
     try:
-        with open(
-            pkg_resources.resource_filename(
-                "language_formatters_pre_commit_hooks",
-                "{tool_name}.version".format(tool_name=tool_name),
-            )
-        ) as f:
+        path = pkg_resources.resource_filename("language_formatters_pre_commit_hooks", f"{tool_name}.version")
```
Comment (macisamuele) [inline language_formatters_pre_commit_hooks/__init__.py:13]: the usage of f-strings will make the code not compatible with Python 3.5

Given that the friction of using `format(...)` approach is limited I would not drop practical usage of the tool on Python 3.5

File: setup.py [hunk 3]
```diff
@@ -1,11 +1,7 @@
-# -*- coding: utf-8 -*-
 from setuptools import setup
 
 
 setup(
-    extras_require={
```
Comment (macisamuele) [inline setup.py:6]: The PR refers to removal of Python2 code , this is still needed for Python3 versions before 3.5

I understand that they are all EOL, but it goes beyond the PR scope and preserving their support is not as expensive

**Review by macisamuele — COMMENTED** [review]
I'm a bit unsure by this PR.
The intent is to drop Python2 support, but while doing it we are dropping support for anything below 3.7.
Hence at the very least the PR title needs to be updated.

The real question for me is: what is the cost of keeping the tool eventually running on Python2 or old version of Python3?
As far as upgrade is good, I would consider what is the cost of allowing users with old Python version to still run the tool. If the cost is not high, then what is the benefit of doing the update?
Is there any feature of the new versions that we cannot use as we would allow old python version to run the code?

**Comment by mxr** [comment]
It's up to you as the maintainer what you want to support. If it's not a burden, then go for it.

In that vein, I would suggest adding all supported versions you support [to the CI matrix](https://github.com/macisamuele/language-formatters-pre-commit-hooks/blob/b0828e7fe7d816fdef78e6f951f18fded3351e2d/.github/workflows/build.yaml#L21), so you can be confident in supporting them properly. Also, it's a best-practice to specify `python_requires>=xxx` in setup.py ([docs](https://packaging.python.org/en/latest/guides/dropping-older-python-versions/#specify-the-version-ranges-for-supported-python-distributions)) if you ever want to remove support for lower versions.

**Pull request closed without merging**